sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -557,9 +585,22 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
                 .switchIfEmpty(Mono.fromFuture(() -> {
                     SilentParameters.SilentParametersBuilder forceParametersBuilder = SilentParameters.builder(
                         new HashSet<>(request.getScopes())).forceRefresh(true);
+
+                    if (request.getClaims() != null) {
+                        ClaimsRequest customClaimRequest = CustomClaimRequest
+                                                               .formatAsClaimsRequest(request.getClaims());
+                        forceParametersBuilder.claims(customClaimRequest);
+                    }
+
                     if (account != null) {
                         forceParametersBuilder = forceParametersBuilder.account(account);
                     }
+
+                    if (request.getClaims() != null) {
```

Comment:
Isn't this already handled above (line 589)? or am I missreading the control scope?